app/scripts/modules/core/src/account/AccountSelectField.tsx

```diff
@@ -1,13 +1,142 @@
+import * as React from 'react';
+import { $q } from 'ngimport';
+import { flatten, has, intersection, isEqual, map, uniq, xor } from 'lodash';
+
 import { IAccount } from 'core/account';
+import { AccountService } from 'core/account/AccountService';
 
 export interface IAccountSelectFieldProps {
   accounts: IAccount[] | string[];
-  component: Object;
+  component: { [key: string]: any };
   field: string;
   provider: string;
   loading?: boolean;
   onChange?: (account: string) => void;
   labelColumns?: number;
   readOnly?: boolean;
-  multiselect?: boolean;
+}
+
+export interface IAccountSelectFieldState {
+  accountContainsExpression: boolean;
+  mergedAccounts: string[];
+  primaryAccounts: string[];
+  secondaryAccounts: string[];
+}
+
+const isExpression = (account: string) => !!account && account.includes('${');
+
+export class AccountSelectField extends React.Component<IAccountSelectFieldProps, IAccountSelectFieldState> {
+  public state: IAccountSelectFieldState = {
+    accountContainsExpression: false,
+    mergedAccounts: [],
+    primaryAccounts: [],
+    secondaryAccounts: [],
+  };
+
+  private groupAccounts = (accounts: IAccount[] | string[]) => {
+    const { component, field, onChange, provider } = this.props;
+
+    if (!accounts || !accounts.length) {
+      return;
+    }
+    if (has(component, field) && isExpression(component[field])) {
+      this.setState({ accountContainsExpression: true });
+      return;
+    }
+
+    const accountsAreObjects = Boolean((accounts[0] as IAccount).name);
+    let getAccountDetails = provider ? AccountService.getAllAccountDetailsForProvider(provider) : $q.when([]);
```

Comment:
looks like an extra api call to `getAllAccountDetailsForProvider` is made then thrown away when `accountsAreObjects === true`

Reply:
Also, looks like normally `getAccountDetails` is a promise for an array of objects.  But when `provider` is truthy and `accountsAreObjects` are falsey, it's a promise for an object not an array of objects?